### ISSUE TITLE
Add code coverage for tests

### ISSUE THREAD

**Issue by zbraniecki** [projectfluent/fluent.js#28]
It would help me to work with the parser tests to know how much of the parser is covered with tests. Istanbul seems like a good candidate? https://github.com/gotwarlost/istanbul

**Comment by stasm** [comment]
I tried running the following commands:

    npm install istanbul
    cd fluent
    ../node_modules/.bin/istanbul cover ../node_modules/.bin/_mocha -- -u tdd -r test/setup.js

which according to https://github.com/gotwarlost/istanbul/issues/44#issuecomment-16093330 _should_ work. Unfortunately I got:

    No coverage information was collected, exit without writing coverage information

**Comment by stasm** [comment]
I'd like to have coverage reports as well. I'll try to get this working today or tomorrow.

**Comment by stasm** [comment]
Looks like `npm` installs istanbul v0.4.5 by default. Installing v1.1.0-alpha.1 fixes the above issue.

I managed to get coverage reports working for all packages except for `fluent-syntax`. I'm getting a weird `ReferenceError: Function is not defined`. I'll dig some more.

**Comment by Pike** [comment]
I'd argue that instanbul isn't a good candidate, looking at the state of releases that project had in the last year (effectively none, and it might be that 1.1.0 alpha 1 is actually older than 0.4.5)

**Comment by stasm** [comment]
Do you know any alternatives? I found `blanket` which isn't maintained anymore; the author recommends `istanbul`.

**Comment by stasm** [comment]
Istanbul is now maintained in https://github.com/istanbuljs and is at version 1.1.2. I gave it another try but ran into a coupe of issues:

  - There's a bug in Istanbul which makes it crash when it sees the `Function` class in `fluent-syntax/src/ast.js`. We could possibly rename that class or exclude that file from the coverage report.

  - The results for `fluent/src/{parser,resolver}.js` looks as if there was unclosed `<span class="cstat-no">` element in the result: it's all red after a point.

@zbraniecki, I don't want to spend too much on this right now. I pushed my WIP branch to https://github.com/stasm/fluent.js/tree/istanbul.  Feel free to take over if you have cycles for this.
